readme.md

```diff
@@ -53,7 +53,71 @@ Directory names should be lowercase. Filenames should be singular.
 *todo: evaluate common prefix like `pagelib` and BEM naming.*
 
 ## Development setup and workflow
-*Coming soon.*
+The Android and iOS Wikipedia apps make use of the `wikimedia-page-library`. At some point their respective build phases invoke `npm install` which causes a *published* version of this library to be retrieved and used in their respective builds. 
+
+But sometimes we want to test *unpublished* `wikimedia-page-library` branches. This could be to review bug fixes or new feature branches or to develop fixes and features ourselves. We'd like to be able to checkout a local branch of `wikimedia-page-library` and run the app's build process and have the app use whatever we had checked out of our local copy of `wikimedia-page-library`.
+
+To do this:
+
+**Step 1** (only need to do this once)
+
+Create a local clone of `wikimedia-page-library`:
+> `git clone https://github.com/wikimedia/wikimedia-page-library.git`
+
+**Step 2** (do this as needed)
+
+Checkout the branch to be tested:
+> `cd ~/wikimedia-page-library/`
+
+> `git checkout somebranch`
+
+**Step 3** (only need to do this once)
+
+Now we need to temporarily tell NPM to use our local copy of `wikimedia-page-library` so when we run the app's build process and it runs `npm install`, our local copy of  `wikimedia-page-library` will be used instead of the published one.
+
+We can use the `npm link` command to do this.
+
+Create a `wikimedia-page-library` symlink:
+> `cd ~/wikimedia-page-library/`
+
+> `npm link`
+
+The `npm link` command above configures NPM with a global `wikimedia-page-library` symlink. Note that it infers the name of the package from the package folder we `cd`'ed into. We will use this symlink in the next step. You only have to create this symlink once as long as you don't change the file system location of your local clone of `wikimedia-page-library`.
+
+**Step 4** (do this when you begin testing against a local copy)
+
+Now we have to configure the app's `www/node_modules/` folder to point to the symlink we just created.
```

Comment:
The manual page for NPM link seems to describe that you don't need to drill into the `node_modules` folder itself, just the repo:

> For example:
>
>         cd ~/projects/node-redis    # go into the package directory
>         npm link                    # creates global link
>         cd ~/projects/node-bloggy   # go into some other package directory.
>         npm link redis              # link-install the package
>
>       Now,  any  changes  to  ~/projects/node-redis  will  be  reflected   in
>       ~/projects/node-bloggy/node_modules/node-redis/.  Note  that  the  link
>       should be to the package name, not the directory name for that package.